bindings.c

```diff
@@ -4525,16 +4625,21 @@ static int proc_loadavg_read(char *buf, size_t size, off_t offset,
 	memcpy(buf, d->buf, total_len);
 	return total_len;
 }
-
+/* Return a positive number on success, return 0 on failure.*/
 pthread_t load_daemon(void)
 {
-	init_load();
 	int ret;
 	pthread_t pid;
+
+	ret = init_load();
+	if (ret == -1) {
+		lxcfs_error("%s\n", "Initialize hash_table fails!");
+		return 0;
+	}
 	ret = pthread_create(&pid, NULL, load_begin, NULL);
```

Comment:
Use of load_begin() should probably be conditional.  Default-on is
ok, but some people may not want to bother with the overhead of
calculation.

Reply:
Do you mean that some people may not want to use __loadavg__? 
I should give an alternative option for these poeple to use loadavg or not.